dynamicio/io/postgres.py

```diff
@@ -130,32 +94,23 @@ def read(self) -> pd.DataFrame:
         else:
             sql_query = self.sql_query
 
-        logger.info(f"Downloading table: {self.final_table_name} from: {self.db_host}:{self.db_name}")
+        logging.info(f"Downloading table: {self.final_table_name} from: {self.db_host}:{self.db_name}")
         with session_scope(self.connection_string, self.application_name) as session:
-            df = pd.read_sql(sql=sql_query, con=session.get_bind(), **self.kwargs)
-
-        df = self.validate(df)
+            df = pd.read_sql(sql=sql_query, con=session.get_bind(), **self.read_kwargs)
 
         return df
 
-        # TODO: sqlalchemy 2.0 breaks here
-        #       session.get_bind() is probably the culprit
-        # TODO: Check if type coercion is needed here
-        #       For example objects -> String(64)s
-        #       For example date goes to Date and datetime64[ns] -> DateTime
-
-    def write(self, df: pd.DataFrame):
+    def _write(self, df: pd.DataFrame) -> None:
         """Handles Write operations for Postgres."""
         if not self.table_name:
-            raise ConfigurationError("PostgresResource must specify table_name for writing.")
-
-        self.check_injections()
-        df = self.validate(df)
+            raise ValueError("PostgresResource must specify table_name for writing.")
 
         with session_scope(self.connection_string, self.application_name) as session:
             session: SqlAlchemySession  # type: ignore # this is done for IDE purposes
             if self.truncate_and_append:
-                logger.info(f"Writing to table (csv-hack): {self.final_table_name} from: {self.db_host}:{self.db_name}")
+                logging.info(
+                    f"Writing to table (csv-hack): {self.final_table_name} from: {self.db_host}:{self.db_name}"
+                )
                 session.execute(f"TRUNCATE TABLE {self.final_table_name};")
```

Comment:
nit: in the future would it make sense to wrap all these steps in a transaction?

Reply:
Should do some testing on this, I don't see why not. ATM I just kept the previous method.